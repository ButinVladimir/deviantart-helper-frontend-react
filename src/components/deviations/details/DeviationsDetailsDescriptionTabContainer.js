import { connect } from 'react-redux';
import DeviationsDetailsDescriptionTab from './DeviationsDetailsDescriptionTab';

/**
 * @description
 * Maps Redux state to props.
 *
 * @param {State} state - The Redux state.
 * @returns {Object} Props.
 */
const mapStateToProps = state => ({
  title: state.deviations.details.title,
  thumbnail: state.deviations.details.thumbnail,
  url: state.deviations.details.url,
  publishedTime: state.deviations.details.publishedTime,
  description: state.deviations.details.description,
  views: state.deviations.details.views,
  favourites: state.deviations.details.favourites,
  comments: state.deviations.details.comments,
  downloads: state.deviations.details.downloads,
});

export default connect(mapStateToProps)(DeviationsDetailsDescriptionTab);
