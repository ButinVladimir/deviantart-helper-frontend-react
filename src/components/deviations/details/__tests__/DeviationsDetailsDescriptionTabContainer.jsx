import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import createDefaultState from '../../../../redux/states/state';
import DeviationsDetailsDescriptionTabContainer from '../DeviationsDetailsDescriptionTabContainer';

describe('DeviationsDetailsDescriptionTabContainer', () => {
  it('can be rendered correctly', () => {
    const state = {
      ...createDefaultState(),
      deviations: {
        details: {
          title: 'Title',
          thumbnail: {
            src: '/path/to/image',
            width: 1,
            height: 2,
          },
          url: 'http://path/to/deviation',
          publishedTime: 1,
          description: 'Description',
          nsfw: true,
          views: 2,
          favourites: 3,
          comments: 4,
          downloads: 5,
        },
      },
    };

    const mockStore = configureStore();
    const store = mockStore(state);
    const wrapper = mount(
      <Provider store={store}>
        <DeviationsDetailsDescriptionTabContainer />
      </Provider>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
