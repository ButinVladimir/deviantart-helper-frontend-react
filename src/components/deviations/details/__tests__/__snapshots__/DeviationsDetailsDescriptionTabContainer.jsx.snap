// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviationsDetailsDescriptionTabContainer can be rendered correctly 1`] = `
Array [
  <section
    className="section"
    style={Object {}}
  >
    <div
      className="container"
      style={Object {}}
    >
      <article
        className="media"
        style={Object {}}
      >
        <div
          className="image-container media-left"
          style={Object {}}
        >
          <div>
            <img
              alt="Title"
              src="/path/to/image"
            />
          </div>
          <div>
            <small
              className="has-text-grey"
            >
              1970-1-1 03:00:00
            </small>
          </div>
          <div>
            <a
              href="http://path/to/deviation"
            >
              Visit DA page
            </a>
          </div>
          <span
            className="tag is-danger"
            style={Object {}}
          >
            NSFW
          </span>
        </div>
        <div
          className="media-content"
          style={Object {}}
        >
          <div
            className="content"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Description",
              }
            }
            style={Object {}}
          />
        </div>
      </article>
    </div>
  </section>,
  <section
    className="section"
    style={Object {}}
  >
    <div
      className="container"
      style={Object {}}
    >
      <div
        className="box"
        style={Object {}}
      >
        <div
          className="level is-tablet"
          style={Object {}}
        >
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  2
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Views
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  3
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Favourites
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  4
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Comments
                </h1>
              </div>
            </div>
          </div>
          <div
            className="level-item has-text-centered"
            style={Object {}}
          >
            <div
              className="content"
              style={Object {}}
            >
              <div>
                <h1
                  className="title is-4"
                >
                  5
                </h1>
              </div>
              <div>
                <h1
                  className="subtitle is-6 is-uppercase"
                >
                  Downloads
                </h1>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
]
`;
